### PR TITLE
Handling large integers

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Parsing large integers breaks in flytectl since by default json unmarshal converts the number to float64 representation
and when fmt.Sprinf is done for such a value then by default for large floats it uses exponential representation which breaks the parsing 
when it tries to do this `strconv.ParseInt("8.888888e+06")`

```
Error: failed to parse integer value: strconv.ParseInt: parsing "8.888888e+06": invalid syntax
```

The fix is to check for the type of float64 and compare the truncated integer portion with the actual float64 value and if they are equal then use %.0f formatter to remove the decimal part in the string.

Test with integer within bounds
```
cat lp.yaml 
iamRoleARN: ""
inputs:
  am: false
  day_of_week: 2
  number: 88888888
kubeServiceAcct: ""
targetDomain: ""
targetProject: ""
version: v1
workflow: core.flyte_basics.lp.go_greet

 flytectl create execution -p flytesnacks -d development --execFile lp.yaml
execution identifier project:"flytesnacks" domain:"development" name:"f586641988abb438294e" 
```

Test with integer out of bounds

```
cat lp.yaml
iamRoleARN: ""
inputs:
  am: false
  day_of_week: 2
  number: 88888888888888888888
kubeServiceAcct: ""
targetDomain: ""
targetProject: ""
version: v1
workflow: core.flyte_basics.lp.go_greet

flytectl create execution -p flytesnacks -d development --execFile lp.yaml
Error: failed to parse integer value: strconv.ParseInt: parsing "88888888888888885248": value out of range
```

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1282

## Follow-up issue
_NA_

### CODE REVIEW

File: clients/go/coreutils/literals.go [hunk 1]
```diff
@@ -492,7 +493,9 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 		if v == nil {
 			strValue = ""
 		}
-
+		if f, ok := v.(float64); ok && math.Trunc(f) == f {
```
Comment (katrogan) [inline clients/go/coreutils/literals.go:496]: mind adding a comment here why this is necessary?

**Review by katrogan — DISMISSED** [review]

File: clients/go/coreutils/literals.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply clients/go/coreutils/literals.go:500]: Done.

**Review by katrogan — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
